src/subsets/Subsets.jsx

```diff
@@ -0,0 +1,263 @@
+                >
+                  <input
+                    type="checkbox"
+                    name={index + checkboxToRender.label}
+                    value={checkboxToRender.label}
+                    className="checkboxInput"
+                    onClick={((e) => this.toggleCheckbox(e, checkboxToRender))}
+                  />
+                  {checkboxToRender.label}
+                </td>
+              );
+            })}
+          </tr>
+          </tbody>
+          </table>
+        );
+
+        checkboxesToRender = [];
+      }
+    });
+
+    return toRender;
+  }
+
+  toggleCheckbox(e,selectedOption) {
+    //console.log(selectedOption)
+    // return this.setState({
+    //   selectedCheckBoxes: selectedCheckBoxes
+    // });
+  }
+
+  handleStateUpdate(val) {
+    this.onStateChange(val);
+    this.props.history.push({
+      pathname: `${this.props.match.url}/${val.value}`
+    });
+  }
+
+  handleMsaMdsUpdate(val) {
+    this.onMsaMdsChange(val);
+    this.props.history.push({
+      pathname: `${this.props.history.location.pathname}/${val.value}`
+    });
+  }
+
+  loadMsaSelect(subsetYear) {
+    let msaMds = stateToMsas[subsetYear][this.state.selectState.value];
+    let msaMdsOptions = msaMds
+      ? msaMds.map(msaMd => {
+          return { value: msaMd.id, label:  msaMd.id+" - "+msaMd.name };
+        })
+      : [{ id: "", value: "" }];
+
+    return (
+      <Select
+        isDisabled={this.state.selectState.value ? false : true}
+        onChange={this.handleMsaMdsUpdate}
+        placeholder="Select MSA/MD..."
+        searchable={true}
+        autoFocus
+        openOnFocus
+        simpleValue
+        options={msaMdsOptions}
+        value={this.state.selectMsaMds}
+      />
+    );
+  }
+
+  loadVarSelect(varOption, varListName) {
+    let varList = variables[varOption][varListName];
+    let varOptions = varList
+      ? varList.map(variableOption => {
+          return { value: variableOption.id, label: variableOption.name };
+        })
+      : [{ id: "", value: "" }];
+    return (
+      <Select
+        isDisabled={this.state.selectState.value ? false : true}
+        onChange={
+          varOption === "1"
+            ? this.handleActionTakenUpdate
+            : this.handleRaceUpdate
+        }
+        placeholder={"Select variable" + varOption + "..."}
+        searchable={true}
+        autoFocus
+        openOnFocus
+        simpleValue
+        options={varOptions}
+        value={
+          varOption === "1"
+            ? this.state.selectActionTaken
+            : this.state.selectRace
+        }
+      />
+    );
+  }
+
+  onStateChange(selectedOption) {
+    return this.setState({
+      selectState: selectedOption,
+      selectMsaMds: "",
+      selectActionTaken: "",
+      selectRace: ""
+    });
+  }
+  onMsaMdsChange(selectedOption) {
+    return this.setState({
+      selectMsaMds: selectedOption
+    });
+  }
+
+  handleActionTakenUpdate(selectedOption) {
```

Comment:
With many variables, the handleActionTakeUpdate and handleRaceUpdate should be the same function that inspects the selected value onChange and updates corresponding state.